### PR TITLE
Fix login form when ldap is enabled but password login is disabled

### LINKED ISSUE #25661: When LDAP is enabled but password login is disabled, login doesn't work
**Describe the bug**
If you set up and enable LDAP, but toggle off the "Enable Password Authentication" setting, the login screen will say "Sign in to Metabase" not present any authentication options.

**To Reproduce**
1. Go to Admin -> Settings -> Authentication
2. Set up and enable LDAP. The easiest way I've found for this is to use [these credentials](https://www.zflexldapadministrator.com/index.php/blog/82-free-online-ldap) for a free online testing server, but you can also run one locally.
3. Scroll down to "Enable Password Authentication" and disable it
4. Log out
5. The login screen will not present any authentication options

**Expected behavior**
I guess we should still show the login form but only attempt to hit the LDAP server when you submit it, not attempt normal email/password login as well

**Screenshots**
Screen Shot 2022-09-27 at 3 28 16 PM

**Information about your Metabase Installation:**
Current master

**Severity**
Pretty serious, it's an easy way to get locked out of an instance. Can be recovered from by setting `MB_ENABLE_PASSWORD_LOGIN=true`.

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/25661
Reproduces https://github.com/metabase/metabase/issues/25661

Please note that there is a separate fix for 0.44 https://github.com/metabase/metabase/pull/25673

How to test:
- Run a test ldap server ` docker run -p 389:389 osixia/openldap:1.5.0`
- Run Metabase EE
- Go to Admin -> Settings -> Authentication -> LDAP and establish connection with the following settings
```
{
    "ldap-enabled": true,
    "ldap-host": "localhost",
    "ldap-port": "389",
    "ldap-bind-dn": "cn=admin,dc=example,dc=org",
    "ldap-password": "admin",
    "ldap-user-base": "dc=example,dc=org",
  }
```
- Go to Admin -> Settings -> Authentication and disable password login
- Log out and make sure the login form is not shown correctly (it's not possible to login with this test ldap server without providing user data)

### CODE REVIEW

File: frontend/src/metabase/admin/people/containers/UserSuccessModal.jsx [hunk 1]
```diff
@@ -35,15 +35,15 @@ class UserSuccessModal extends React.Component {
         {temporaryPassword ? (
           <PasswordSuccess user={user} temporaryPassword={temporaryPassword} />
         ) : (
-          <EmailSuccess isEeSsoConfigured={isSsoConfigured} user={user} />
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/people/containers/UserSuccessModal.jsx:38]: Ahhh. Nice catch. I must have missed that from the last time I refactored this file.

File: frontend/src/metabase/auth/components/PasswordButton/PasswordButton.tsx [hunk 2]
```diff
@@ -2,16 +2,26 @@ import React from "react";
...
+}: PasswordButtonProps) => {
   const link = redirectUrl
     ? `/auth/login/password?redirect=${encodeURIComponent(redirectUrl)}`
     : `/auth/login/password`;
 
-  return <AuthButton link={link}>{t`Sign in with email`}</AuthButton>;
+  return (
+    <AuthButton link={link}>
+      {isLdapEnabled
+        ? t`Sign in with username or email`
```
Comment (WiNloSt) [inline frontend/src/metabase/auth/components/PasswordButton/PasswordButton.tsx:21]: Not sure about the copy update if it's been discussed somewhere.

**Review by WiNloSt — APPROVED** [review]
Works as described 👍

File: frontend/src/metabase/auth/components/PasswordButton/PasswordButton.tsx (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/auth/components/PasswordButton/PasswordButton.tsx:21]: This is to mirror the following logic https://github.com/metabase/metabase/blob/b50b06ecce3c1a6992a7194a8c0da51c948f7191/frontend/src/metabase/entities/users/forms.js#L141

**Pull request merged**